### PR TITLE
videocomposer: clip refactor

### PR DESCRIPTION
**Pull request by wtomin**
Sorry that I accidentally pushed this branch to mindspore-lab: mindone. I will delete this branch after merging this commit.

clip refactor:
- [x] remove the clip codes in videocomposer, reuse the stable_diffusion_v2/tools/_common/clip;
- [x] modified CLIPTokenizer to process empty strings, like "".
- [x] change the import path is videocomposer/vc/models/clip_embedder.py

### CODE REVIEW

File: examples/stable_diffusion_v2/tools/_common/clip/clip_tokenizer.py [hunk 1]
```diff
@@ -625,8 +625,10 @@ def tokenize(self, text):
 
     def _convert_tokens_to_ids(self, input_tokens):
         r"""Convert_tokens_to_ids"""
-        if not input_tokens:
+        if input_tokens is None:
```
Comment (DongyaoZhu) [inline examples/stable_diffusion_v2/tools/_common/clip/clip_tokenizer.py:628]: please confirm a change of behavior on input="" (empty string)

**Review by DongyaoZhu — CHANGES_REQUESTED** [review]
weak reject

**Comment by wtomin** [comment]
> weak reject

Before change, clip_tokenizer("") will raise an error: `ValueError("The length of the token_ids should be larger than 0.")`.
After change:
```
$ python
Python 3.7.16 (default, Jan 17 2023, 15:51:26) 
[GCC 11.2.0] :: Anaconda, Inc. on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> from tools._common.clip import CLIPTokenizer
>>> text_processor = CLIPTokenizer("ldm/models/clip/bpe_simple_vocab_16e6.txt.gz", pad_token="!")
>>> text_processor("")
{'input_ids': [49406, 49407], 'attention_mask': [1, 1]}
>>> text_processor("", add_special_tokens=False)
{'input_ids': [], 'attention_mask': []}
>>> text_processor("", add_special_tokens=False, max_length=77, padding='max_length')
{'input_ids': [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0], 'attention_mask': [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0]}
>>> text_processor("", add_special_tokens=True, max_length=77, padding='max_length')
{'input_ids': [49406, 49407, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0], 'attention_mask': [1, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0]}
```

**Comment by DongyaoZhu** [comment]
> > weak reject
> 
> Before change, clip_tokenizer("") will raise an error: `ValueError("The length of the token_ids should be larger than 0.")`. After change:
> 
> ```
> $ python
> Python 3.7.16 (default, Jan 17 2023, 15:51:26) 
> [GCC 11.2.0] :: Anaconda, Inc. on linux
> Type "help", "copyright", "credits" or "license" for more information.
> >>> from tools._common.clip import CLIPTokenizer
> >>> text_processor = CLIPTokenizer("ldm/models/clip/bpe_simple_vocab_16e6.txt.gz", pad_token="!")
> >>> text_processor("")
> {'input_ids': [49406, 49407], 'attention_mask': [1, 1]}
> >>> text_processor("", add_special_tokens=False)
> {'input_ids': [], 'attention_mask': []}
> >>> text_processor("", add_special_tokens=False, max_length=77, padding='max_length')
> {'input_ids': [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0], 'attention_mask': [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0]}
> >>> text_processor("", add_special_tokens=True, max_length=77, padding='max_length')
> {'input_ids': [49406, 49407, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0], 'attention_mask': [1, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0]}
> ```

oh you aim to do this? ok raised

**Review by DongyaoZhu — APPROVED** [review]

**Comment by wtomin** [comment]
Yes, I need tokenizer("") to return [bos_token_id, eos_token_id], which is aligned with openclip tokenizer's behavior.

**Review by SamitHuang — APPROVED** [review]

**Pull request merged**